### PR TITLE
Add a version of timing_step_37 using MGTransferMF

### PR DESCRIPTION
**Pull request by peterrum**
~depends on #15794~

### CODE REVIEW

**Comment by peterrum** [comment]
I have rebased this PR.

File: tests/performance/timing_step_37_gc.cc [hunk 1]
```diff
@@ -0,0 +1,675 @@
...
+          {"setup_grid",
+           "setup_dofs",
+           "setup_matrix_free",
+           "assemble_rhs",
+           "setup_transfer",
+           "setup_smoother",
+           "solve",
+           "matvec_double",
+           "matvec_float",
+           "matvec_float_coarser"}};
```
[the remark this answers is not in the document]
Comment (kronbichler) [reply tests/performance/timing_step_37_gc.cc:667]: I agree, we should only measure the setup of the transfer and the solver time. Everything else should be equivalent I believe.

File: tests/performance/timing_step_37_gc.cc [hunk 2]
```diff
@@ -0,0 +1,621 @@
...
+
+  debug_output << std::endl;
+  return {timer["setup_transfer"].wall_time(), timer["solve"].wall_time()};
+}
+
+
+std::tuple<Metric, unsigned int, std::vector<std::string>>
+describe_measurements()
+{
+  return {Metric::timing, 4, {"setup_transfer", "solve"}};
```
Comment (peterrum) [inline tests/performance/timing_step_37_gc.cc:613]: Done!

**Comment by kronbichler** [comment]
@tamiko any last opinions on this PR? If not, I will merge tomorrow.

**Pull request merged**